### PR TITLE
feat: Adding support for Ton

### PR DESCRIPTION
**Pull request by y3fers0n**
```
    ___       ___       ___   
   /\  \     /\  \     /\__\  
   \:\  \   /::\  \   /:| _|_ 
   /::\__\ /:/\:\__\ /::|/\__\
  /:/\/__/ \:\/:/  / \/|::/  /
  \/__/     \::/  /    |:/  / 
             \/__/     \/__/  
```
## Adding support for Ton Bridge & Provider

`@trustwallet/web3-provider-ton`

### Supported methods
- [x] tonConnect_connect 
- [x] tonConnect_reconnect
- [x] ton_rawSign -> signMessage
- [x] tonConnect_sendTransaction -> signTransaction
- [x] ton_sendTransaction
- [x] ton_requestAccounts
- [x] ton_requestWallets
- [x] ton_getBalance: should be supported by clients

### Not supported yet
- [ ] ton_decryptMessage
- [ ]  ton_encryptMessage

Screenshot 2024-07-29 at 5 32 34 PM

### CODE REVIEW

File: packages/ton/TonBridge.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+/**
+ * Ton bridge implementation
+ *
+ * Based on https://docs.ton.org/develop/dapps/ton-connect/protocol & open mask
+ */
+export class TonBridge implements TonConnectBridge {
+  deviceInfo!: DeviceInfo;
+  walletInfo?: WalletInfo | undefined;
+  protocolVersion: number = 2;
```
Comment (DaStepn) [inline packages/ton/TonBridge.ts:36]: Do we have to keep monitoring https://docs.ton.org/develop/dapps/ton-connect/protocol to make sure this value is aligned with whatever changes they make on their end?

**Review by DaStepn — APPROVED** [review]

File: packages/ton/TonBridge.ts (hunk 1 above)
Comment (y3fers0n) [reply packages/ton/TonBridge.ts:36]: Yes, if there are new changes to the protocol we can increase the number when we support those changes. This indicates max supported Ton Connect version

**Review by Jjajsja — COMMENTED** [review]
Implementación de cambios

**Comment by ronaldoguedess** [comment]
👏🏻👏🏻👏🏻 Amazing!
Is already possible to use? Or, do you need to publish a new version first?
I use java

**Pull request merged**